### PR TITLE
OGM-913 Allows join on collection of elements

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-913

This only works for Neo4j and MongoDB at the moment, I'll create a new issue for the lucene parser since it requires some changes in the HQL parser project

### CODE REVIEW

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/impl/MongoDBPropertyHelper.java [hunk 1]
```diff
@@ -36,6 +43,27 @@ public String getColumnName(String entityType, List<String> propertyPath) {
...
+			return ( (CollectionType) propertyType ).getElementType( sessionFactory );
+		}
+		return propertyType;
+	}
+
+	@Override
+	public Object convertToBackendType(String entityType, List<String> propertyPath, Object value) {
+		Type propertyType = getPropertyType( entityType, propertyPath );
+		if ( isElementCollection( propertyType ) ) {
+			// For colleciton of elements we return the type of the collection
```
Comment (gsmet) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/impl/MongoDBPropertyHelper.java]: s/colleciton/collection/

**Comment by DavideD** [comment]
@gsmet I'll reorder the commits. I usually do that but I guess I forgot this time

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/impl/MongoDBPropertyHelper.java (hunk 1 above)
Comment (DavideD) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/impl/MongoDBPropertyHelper.java]: Fixed

**Comment by DavideD** [comment]
I fixed the commits and pushed-forced.

The commits should be in the right order, first the fix and then the test.

**Comment by gsmet** [comment]
@DavideD ok, thanks. I'm not very confident in the order GitHub displays the commits, especially after a force push, so I prefer if you push it directly when you're done!

Thanks!

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/827

**Pull request merged**